crates/rooch-key/src/key_derive.rs

```diff
@@ -81,95 +81,95 @@ impl CoinOperations<EthereumAddress, Secp256k1RecoverableKeyPair> for CoinID {
-                Some(p) => {
-                    // The derivation path must be hardened at all levels with purpose = 44, coin_type = 784
-                    if let &[purpose, coin_type, account, change, address] = p.as_ref() {
-                        if Some(purpose)
-                            == ChildNumber::new(DERVIATION_PATH_PURPOSE_ECDSA, true).ok()
-                            && Some(coin_type)
-                                == ChildNumber::new(
-                                    DERIVATION_PATH_COIN_TYPE_SUI.try_into().unwrap(),
-                                    true,
-                                )
-                                .ok()
-                            && account.is_hardened()
-                            && change.is_hardened()
-                            && address.is_hardened()
-                        {
-                            Ok(p)
-                        } else {
-                            Err(RoochError::SignatureKeyGenError("Invalid path".to_owned()))
-                        }
+    } else {
+        // Rooch chain id
+        match path {
+            Some(p) => {
+                // The derivation path must be hardened at all levels with purpose = 44, coin_type = 784
+                if let &[purpose, coin_type, account, change, address] = p.as_ref() {
+                    if Some(purpose) == ChildNumber::new(DERVIATION_PATH_PURPOSE_ED25519, true).ok()
+                        && Some(coin_type)
+                            == ChildNumber::new(
+                                DERIVATION_PATH_COIN_TYPE_SUI.try_into().unwrap(),
+                                true,
+                            )
+                            .ok()
+                        && account.is_hardened()
+                        && change.is_hardened()
+                        && address.is_hardened()
+                    {
+                        Ok(p)
                     } else {
                         Err(RoochError::SignatureKeyGenError("Invalid path".to_owned()))
                     }
+                } else {
+                    Err(RoochError::SignatureKeyGenError("Invalid path".to_owned()))
                 }
-                None => Ok(format!(
-                    "m/{DERVIATION_PATH_PURPOSE_ECDSA}'/{DERIVATION_PATH_COIN_TYPE_SUI}'/0'/0'/0'"
-                )
-                .parse()
-                .map_err(|_| RoochError::SignatureKeyGenError("Cannot parse path".to_owned()))?),
             }
+            None => Ok(format!(
+                "m/{DERVIATION_PATH_PURPOSE_ED25519}'/{DERIVATION_PATH_COIN_TYPE_SUI}'/0'/0'/0'"
+            )
+            .parse()
+            .map_err(|_| RoochError::SignatureKeyGenError("Cannot parse path".to_owned()))?),
         }
-        CoinID::Bitcoin => todo!(),
-        CoinID::Nostr => todo!(),
     }
 }
 
 pub fn generate_new_key_pair<Addr, KeyPair>(
-    coin_id: CoinID,
+    multichain_id: RoochMultiChainID,
     derivation_path: Option<DerivationPath>,
```

Comment:
We do not need to support every chain's keypair. We can replace the multichain id with a key pair type.

Reply:
I defined a `KeyPairType` and assign every key pairs a type either of `evm` or `rooch`.